ESP32 Porting GSM Module Compile Error

```
[1/8] Linking C executable zephyr\zephyr_prebuilt.elf
FAILED: zephyr/zephyr_prebuilt.elf zephyr/zephyr_prebuilt.map
cmd.exe /C "cd . && C:\Users\Colt\.espressif\tools\xtensa-esp32-elf\esp-2020r3-8.4.0\xtensa-esp32-elf\bin\xtensa-esp32-elf-gcc.exe   zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj -o zephyr\zephyr_prebuilt.elf  -Wl,-T  zephyr/linker_zephyr_prebuilt.cmd  -Wl,-Map=D:/ZephyrProject/code/zephyrproject/zephyr/build/zephyr/zephyr_prebuilt.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/xtensa/core/libarch__xtensa__core.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/posix/liblib__posix.a  zephyr/subsys/net/libsubsys__net.a  zephyr/subsys/net/l2/ppp/libsubsys__net__l2__ppp.a  zephyr/subsys/net/ip/libsubsys__net__ip.a  zephyr/subsys/net/lib/conn_mgr/libsubsys__net__lib__conn_mgr.a  zephyr/subsys/net/lib/dns/libsubsys__net__lib__dns.a  zephyr/subsys/random/libsubsys__random.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/entropy/libdrivers__entropy.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  zephyr/CMakeFiles/offsets.dir/./arch/xtensa/core/offsets/offsets.c.obj  -L"c:/users/colt/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.4.0"  -LD:/ZephyrProject/code/zephyrproject/zephyr/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -no-pie  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -lhal  -LD:/ZephyrProject/code/zephyrproject/modules/hal/espressif/zephyr/../components/xtensa/esp32 && cmd.exe /C "cd /D D:\ZephyrProject\code\zephyrproject\zephyr\build\zephyr && "C:\Program Files\CMake\bin\cmake.exe" -E echo ""
c:/users/colt/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: zephyr/subsys/net/l2/ppp/libsubsys__net__l2__ppp.a(ppp_l2.c.obj):(.literal.ppp_recv+0x0): undefined reference to `_ppp_protocol_handler_list_start'
c:/users/colt/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: zephyr/subsys/net/l2/ppp/libsubsys__net__l2__ppp.a(ppp_l2.c.obj):(.literal.ppp_recv+0x4): undefined reference to `_ppp_protocol_handler_list_end'
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' --build 'D:\ZephyrProject\code\zephyrproject\zephyr\build'
```

who can help me？

--- a/soc/xtensa/esp32/linker.ld
+++ b/soc/xtensa/esp32/linker.ld
@@ -265,41 +265,6 @@ _net_buf_pool_list = _esp_net_buf_pool_list;
  SECTION_PROLOGUE(_RODATA_SECTION_NAME,,ALIGN(20))
   {
     _rodata_start = ABSOLUTE(.);
-#if defined(CONFIG_NET_SOCKETS)
-    . = ALIGN(4);
-    Z_LINK_ITERABLE(net_socket_register);
-    . = ALIGN(4);
-#endif
-
-#if defined(CONFIG_SETTINGS)
-    . = ALIGN(4);
-    Z_LINK_ITERABLE(settings_handler_static);
-    . = ALIGN(4);
-#endif
-
-#if defined(CONFIG_BT_BREDR)
-    . = ALIGN(4);
-    Z_LINK_ITERABLE(bt_l2cap_br_fixed_chan);
-    . = ALIGN(4);
-#endif
-
-#if defined(CONFIG_BT_MESH)
-    Z_LINK_ITERABLE(bt_mesh_subnet_cb);
-    Z_LINK_ITERABLE(bt_mesh_app_key_cb);
-    Z_LINK_ITERABLE(bt_mesh_hb_cb);
-#endif
-
-    . = ALIGN(4);
-    Z_LINK_ITERABLE(bt_gatt_service_static);
-    . = ALIGN(4);
-
-    . = ALIGN(4);
-    Z_LINK_ITERABLE(bt_l2cap_fixed_chan);
-    . = ALIGN(4);
-
-    . = ALIGN(4);
-    Z_LINK_ITERABLE(shell);
-    . = ALIGN(4);
 
     __esp_shell_root_cmds_start = .;
     KEEP(*(SORT(.shell_root_cmd_*)));
@@ -337,10 +302,7 @@ _net_buf_pool_list = _esp_net_buf_pool_list;
     _rodata_end = ABSOLUTE(.);
   } GROUP_LINK_IN(ROMABLE_REGION)
 
-#pragma push_macro("Z_ITERABLE_SECTION_ROM")
 #pragma push_macro("ROMABLE_REGION")
-#undef Z_ITERABLE_SECTION_ROM
-#define Z_ITERABLE_SECTION_ROM(x,y)
 #undef ROMABLE_REGION
 /* This is to workaround limitation of `esptool` which needs single `FLASH` data segment
  * which is already defined above. In case, `common-rom.ld` creates additional segments
@@ -355,7 +317,6 @@ __log_backends_end = __esp_log_backends_end;
 __shell_root_cmds_start = __esp_shell_root_cmds_start;
 __shell_root_cmds_end = __esp_shell_root_cmds_end;
 #pragma pop_macro("ROMABLE_REGION")
-#pragma pop_macro("Z_ITERABLE_SECTION_ROM")
 
   SECTION_PROLOGUE(_TEXT_SECTION_NAME, , ALIGN(4))
   {

--- a/west.yml
+++ b/west.yml
@@ -42,7 +42,7 @@ manifest:
       path: modules/lib/civetweb
     - name: hal_espressif
       west-commands: west/west-commands.yml
-      revision: 90952df320f30c696ea73a22a0e50f20b8f94fd3
+      revision: 086d305202390589a206523c239e604718759402
       path: modules/hal/espressif
     - name: fatfs
       revision: 1d1fcc725aa1cb3c32f366e0c53d7490d0fe1109

